client/landing/stepper/declarative-flow/update-design.ts

```diff
@@ -0,0 +1,64 @@
+import { translate } from 'i18n-calypso';
+import {
+	setSignupCompleteSlug,
+	persistSignupDestination,
+	setSignupCompleteFlowName,
+} from 'calypso/signup/storageUtils';
+import { useQuery } from '../hooks/use-query';
+import { useSiteSlug } from '../hooks/use-site-slug';
+import { recordSubmitStep } from './internals/analytics/record-submit-step';
+import DesignSetup from './internals/steps-repository/design-setup';
+import Processing from './internals/steps-repository/processing-step';
+import { ProvidedDependencies } from './internals/types';
+import type { Flow } from './internals/types';
+
+const updateDesign: Flow = {
+	name: 'update-design',
+	get title() {
+		return translate( 'Choose Design' );
+	},
+	useSteps() {
+		return [
+			{ slug: 'designSetup', component: DesignSetup },
+			{ slug: 'processing', component: Processing },
+		];
+	},
+
+	useStepNavigation( currentStep, navigate ) {
+		const flowName = this.name;
+		const siteSlug = useSiteSlug();
+		const flowToReturnTo = useQuery().get( 'flowToReturnTo' ) || 'free';
+
+		function submit( providedDependencies: ProvidedDependencies = {} ) {
+			recordSubmitStep( providedDependencies, 'update-design', flowName, currentStep );
+			switch ( currentStep ) {
+				case 'processing':
+					return window.location.assign(
+						`/setup/${ flowToReturnTo }/launchpad?siteSlug=${ siteSlug }`
+					);
+
+				case 'designSetup':
+					if ( providedDependencies?.goToCheckout ) {
+						const destination = `/setup/${ flowToReturnTo }/launchpad?siteSlug=${ providedDependencies.siteSlug }`;
+						persistSignupDestination( destination );
+						setSignupCompleteSlug( providedDependencies?.siteSlug );
```

Comment:
Do we need to set `setSignupCompleteSlug` and `setSignupCompleteFlowName`?

Reply:
@jeyip - I copied this logic from the free flow [here](https://github.com/Automattic/wp-calypso/blob/00f4e78eb45b97ea108c495d9b206b13e03443c7/client/landing/stepper/declarative-flow/free.ts#L98). 

I considered that it may not be needed here, as this flow is accessed later in the onboarding flow than the original designSetup step. 

But this logic is here to handle if a user upgrades to a premium theme. It seemed plausible that a user would pick a free theme the first time, and then when changing themes here, pick a premium them. So they'd still need to be routed through the same upgrade/cart process. Thus leaving all that logic here and intact. 

Let me know if you have other concerns.